core/Archive/ArchiveState.php

```diff
@@ -102,19 +111,19 @@ private function checkArchiveStates(
 
     private function checkTsArchived(
         ?string $state,
-        Site $site,
-        Range $range,
+        string $siteTimezone,
+        string $periodEndDay,
         string $tsArchived
     ): ?string {
         if (self::COMPLETE !== $state) {
             // only archives detected as complete can be archived before range end
             return $state;
         }
 
-        $rangeEndTimestamp = $range->getDateTimeEnd()->setTimezone($site->getTimezone())->getTimestamp();
-        $tsArchivedTimestamp = Date::factory($tsArchived)->getTimestamp();
+        $datePeriodEnd = Date::factory($periodEndDay . ' 23:59:59')->setTimezone($siteTimezone);
+        $dateArchived = Date::factory($tsArchived);
 
-        if ($tsArchivedTimestamp <= $rangeEndTimestamp) {
+        if ($dateArchived->isEarlier($datePeriodEnd)) {
```

Comment:
Is this change a problem? `isEarlier` is `<` while you are replacing `<=` are we losing anything here?

Reply:
Good catch, this would have indeed already defined an archive as complete at `:59:59`, instead of `:00:00`. Have also updated the tests to take taht single second difference into account.